module/Finna/src/Finna/RecordDriver/SolrForward.php

```diff
@@ -247,18 +247,153 @@ class SolrForward extends \VuFind\RecordDriver\SolrDefault
     ];
 
     /**
-     * Record metadata
+     * Identification strings for where should the author be saved in the results
```

Comment:
"Mappings from FORWARD author type and role to the results" ?

Reply:
Tässä alkaa olla aika monta array:ta, jotka vaikuttavat siihen, miten tekijät mapataan. Tulee mieleen, pitäisikö olla ennemmin yksi iso array tyyliin:

```
    protected $authorConfig = [
        'nonPresenters' => [
            'relators' => [ ... /* entinen $nonPresenterAuthorRelators */ ],
            'mappings' => [ /* ... */],
        ],
        'presenters' => [
            'relators' => [ ... /* entinen $presenterAuthorRelators */ ],
            'mappings' => [ // type => role => credited/uncredited => mapping
                'elonet_henkilo' => [
                    'act' => [
                        'credited' => ['presenters' => 'credited'],
                        'uncredited' => ['presenters' => 'uncredited'],
                    ],
                    'no_value' => [
                        'credited' => ['presenters' => 'performer'],
                        'uncredited' => ['presenters' => 'uncreditedPerformer'],
                    ]
                ],
                'elonet_kokoonpano' => [
                    'no_value' => [
                        'credited' => ['presenters' => 'performingEnsemble'],
                    ],
                    'default' => [ 
                        'credited' => ['presenters' => 'actingEnsemble'],
                    ],
                ],
                /* ... */
            ]
        ],
    ];

```

'default' olisi se, johon menisi, jos ei muuta löydy. Näin tuo konffi olisi helpompi ymmärtää, ja mappauskin olisi selkeämpi koodissa.

Reply:
Tää kuulostaa kyllä järkevältä.